pkg/k8s/apis/cilium.io/v2/client/register.go

```diff
@@ -182,6 +182,20 @@ func createCCNPCRD(clientset apiextensionsclient.Interface) error {
 		CRDName = CustomResourceDefinitionPluralName + "." + SchemeGroupVersion.Group
 	)
 
+	CCNPCRV := CNPCRV.DeepCopy()
+	clusterwideSpec := CCNPCRV.OpenAPIV3Schema.Properties["spec"]
+	clusterwideSpec.Required = nil
+	clusterwideSpec.OneOf = []apiextensionsv1beta1.JSONSchemaProps{
+		{
+			Required: []string{"endpointSelector"},
+		},
+		{
+			Required: []string{"nodeSelector"},
+		},
+	}
+	CCNPCRV.OpenAPIV3Schema.Properties["spec"] = clusterwideSpec
```

Comment:
I think the `"specs"` also needs to be updated.

Reply:
Uh, I keep forgetting things today...